cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunTestsTask.java

```diff
@@ -432,4 +458,11 @@ private ModuleStatus loadModuleStatusFromFile(Path statusJsonPath) throws IOExce
         BufferedReader bufferedReader = Files.newBufferedReader(statusJsonPath, StandardCharsets.UTF_8);
         return gson.fromJson(bufferedReader, ModuleStatus.class);
     }
+
+    private List<String> readFailedTestsFromFile(Path rerunTestJsonPath) throws IOException {
+        Gson gson = new Gson();
+        rerunTestJsonPath = Paths.get(rerunTestJsonPath.toString(), "rerun_test.json");
+        BufferedReader bufferedReader = Files.newBufferedReader(rerunTestJsonPath, StandardCharsets.UTF_8);
```

Comment:
We should catch the exception and close the `BufferedReader`. Lets wrap this with a try-with-resources block.

Reply:
Used `try-with-resource` block and handled exceptions within the function itself